lib/list.js

```diff
@@ -67,8 +67,14 @@ function List(items, itemType, parent) {
     if (isClass(this.itemType)) {
       return new this.itemType(item);
     } else {
-      if (Array.isArray(item)) return item;
-      else return this.itemType(item);
+      if (Array.isArray(item)) {
+        return item;
+      } else if (this.itemType === Date) {
+        if (item === null) return null;
```

Comment:
FYI keeping the same conversion we do in https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/model-builder.js#L704-L708 for non-array Date properties here.